appengine-java-logging/src/test/java/com/google/apphosting/logging/JsonCaptureHandler.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.apphosting.logging;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+
+public class JsonCaptureHandler extends Handler {
+
+  private JsonFormatter formatter;
+  private List<String> events;
+
+  public JsonCaptureHandler() {
+    super();
+    formatter = new JsonFormatter();
+    events = new ArrayList<>();
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    events.add(formatter.format(record));
+  }
+
+  @Override
+  public void flush() {
+  }
+
+  @Override
+  public void close() throws SecurityException {
```

Comment:
Yes, but as subclass you don't have to specify it if you are not actually throwing it.